packages/sdk/src/Config.ts

```diff
@@ -0,0 +1,64 @@
+import { BaseProvider } from '@ethersproject/providers';
+import {
+  Environment,
+  ConfigDomains,
+  EthereumConfig,
+  SubstrateConfig,
+  Resource,
+} from 'types/config';
+
+export class Config {
+  private devnetConfigUrl = 'https://config.develop.buildwithsygma.com/share/';
+  private testnetConfigUrl = 'https://config-server.test.buildwithsygma.com/share/';
+  private mainnetConfigUrl = '';
+
+  public environment!: ConfigDomains;
+  public provider!: BaseProvider;
+
+  public async init(environment: Environment, provider: BaseProvider): Promise<void> {
+    this.provider = provider;
+
+    let network;
+    switch (environment) {
+      case Environment.DEVNET: {
+        network = this.devnetConfigUrl;
+        break;
+      }
+      case Environment.TESTNET: {
+        network = this.testnetConfigUrl;
+        break;
+      }
+      default:
+        network = this.mainnetConfigUrl;
+    }
+
+    try {
+      const response = await fetch(network);
+      this.environment = (await response.json()) as unknown as ConfigDomains;
+    } catch (err) {
+      if (err instanceof Error) {
+        throw new Error(`Failed to fetch shared config because of: ${err.message}`);
+      } else {
+        throw new Error('Failed to fetch shared config');
+      }
+    }
+  }
+
+  public getDomains(): Array<EthereumConfig | SubstrateConfig> {
```

Comment:
This should return just the `Domain` type and not specific configs.

Reply:
are the domains uniformed enough to just return the base type? If so I agree with @mpetrun5 but I have the impression that `etherumconfig` will return an object with much more filled properties than a `substrate` one

Reply:
It shouldn't be this object anyway because transfer will accept just the simple `Domain` object with id and name.

Reply:
but this looks like a generic function to just get the domain, why not return the complete domain? Unless in the general workflow, this will be used internally for the purposes you describe. Then I ageee.

Reply:
This will be used for the public purposes to display supported domains so user can either check what is supported or match the wanted chainID/parachainID to the wanted to chain and add that simple object to the `Transfer`.
Because Transfer contains 
`
from: Domain,
to: Domain
`